syntax = "proto3";

package transport_catalogue_serialize;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string name = 1;
    Coordinates coords = 2;
    map<string, uint32 > road_distances = 3;
}

message Bus {
    string name = 1;
    repeated string route = 2;
    bool is_roundtrip = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
}

message SerializationSettings {
    string filename = 1;
}

message DataBase {
    TransportCatalogue tc = 1;
    RenderSettings render_settings = 2;
    RoutingSettings route_settings = 3;
}